plugin/storage/es/factory.go

```diff
@@ -280,13 +280,18 @@ var _ io.Closer = (*Factory)(nil)
 // Close closes the resources held by the factory
 func (f *Factory) Close() error {
 	var errs []error
+
 	for _, w := range f.watchers {
 		errs = append(errs, w.Close())
 	}
 	if cfg := f.Options.Get(archiveNamespace); cfg != nil {
 		errs = append(errs, cfg.TLS.Close())
 	}
-	errs = append(errs, f.Options.GetPrimary().TLS.Close())
+	errs = append(errs,
+		f.Options.GetPrimary().TLS.Close(),
+		f.getArchiveClient().Close(),
+		f.getPrimaryClient().Close())
```

Comment:
I was thinking this would need to be more complicated, but I think you're right, as long as we close the watchers above, there's no more concurrency with the clients being swapped, so it's safe to close them like this.
👍